### PR TITLE
Dashboard Performance: Switch approach to static method call/preloading

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Mantis Issue: https://mantis.ilias.de/view.php?id=31601

### CODE REVIEW

**Comment by alex40724** [comment]
@mjansenDatabay I wanted to assign Ingmar to this one, so he can decide. However I was not able to assign @iszmais . Not sure why...

File: Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockMembershipsProvider.php [hunk 1]
```diff
@@ -37,59 +37,101 @@ public function __construct(ilObjUser $actor)
...
     }
 
     /**
      * Gets all objects the current user is member of
-     * @param string[] $objTypes
+     * @param array $types
      * @return array array of objects
      */
-    protected function getObjectsByMembership(array $objTypes = []) : array
+    protected function getObjectsByMembership($types = array())
```
Comment (iszmais) [inline Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockMembershipsProvider.php]:
```suggestion
    protected function getObjectsByMembership(array $types = []) : array
```

File: Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockMembershipsProvider.php [hunk 2]
```diff
@@ -37,59 +37,101 @@ public function __construct(ilObjUser $actor)
...
     {
-        if (!is_array($objTypes) || $objTypes === []) {
-            $objTypes = $this->repository->getValidObjectTypes();
-        }
+        $items = array();
```
Comment (iszmais) [inline Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockMembershipsProvider.php]:
```suggestion
        $items = [];
```

File: Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockMembershipsProvider.php [hunk 3]
```diff
@@ -37,59 +37,101 @@ public function __construct(ilObjUser $actor)
...
+                        break;
+                }
             }
+        } else {
+            $crs_mbs = ilParticipants::_getMembershipByType($this->actor->getId(), 'crs');
+            $grp_mbs = ilParticipants::_getMembershipByType($this->actor->getId(), 'grp');
+            $items = array_merge($crs_mbs, $grp_mbs);
+        }
+
+        $references = array();
```
Comment (iszmais) [inline Services/Dashboard/ItemsBlock/classes/class.ilPDSelectedItemsBlockMembershipsProvider.php]:
```suggestion
        $references = [];
```

**Review by iszmais — COMMENTED** [review]
@mjansenDatabay thx for the suggestion!

Just 3 very minor points from my side :sweat_smile: just some code convetion, guess this should be instand agreable.

Greetings,
@iszmais

**Comment by iszmais** [comment]
Thx LGTM :+1: 

@lscharmer Pls take a review for yourself about the part of the Dashboard. Feel free to contact me if you arent familiar with the problem.

@smeyer-ilias I would appreciate your approval for the parts outside the dashboard.

**Comment by lscharmer** [comment]
After reading the mantis ticket I think this PR is unnecessary, because a valid solution is already merged: https://github.com/ILIAS-eLearning/ILIAS/pull/3775

**Comment by iszmais** [comment]
Thx @lscharmer !
wasnt fully aware of the efffect of this but after short testing by myself i agree that this should be sufficent for the reported case in mantis.

In conclusion i will close this pr.

Greeting,
@iszmais

**Pull request closed without merging**